src/middlewared/middlewared/plugins/account.py

```diff
@@ -320,18 +320,13 @@ async def query(self, filters, options):
         )
 
     @private
-    def validate_homedir_mountinfo(self, verrors, schema, dev):
-        mntinfo = self.middleware.call_sync(
-            'filesystem.mount_info',
-            [['device_id.dev_t', '=', dev]],
-            {'get': True}
-        )
-
-        if 'RO' in mntinfo['mount_opts']:
+    def validate_homedir_mountinfo(self, verrors, schema, home_path):
```

Comment:
`dev` and `home_path` are the same?? Seems strange that we were passing a `pathlib.Path` object into the filters of `filesystem.mount_info`? How did this work at all?

Reply:
Originally st_dev was being passed, not a pathlib.Path.